x-pack/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/alert_context_menu.tsx

```diff
@@ -43,25 +43,25 @@ import { useEventFilterAction } from './use_event_filter_action';
 import { useAddToCaseActions } from './use_add_to_case_actions';
 import { isAlertFromEndpointAlert } from '../../../../common/utils/endpoint_alert_check';
 import type { Rule } from '../../../../detection_engine/rule_management/logic/types';
+import { useOpenAlertDetailsAction } from './use_open_alert_details';
 
 interface AlertContextMenuProps {
   ariaLabel?: string;
   ariaRowindex: number;
   columnValues: string;
   disabled: boolean;
   ecsRowData: Ecs;
-  refetch: inputsModel.Refetch;
   onRuleChange?: () => void;
   scopeId: string;
 }
 
+// eslint-disable-next-line complexity
```

Comment:
You won the unlucky lottery with all the existing code in this function that uses `?` chaining, but one way to remove the new `// eslint-disable-next-line complexity` is to extract

```ts
const alertId = ecsRowData?.kibana?.alert ? ecsRowData?._id : null;
```

on line 78 into a separate function, e.g.

```ts
const getAlertId = (ecsRowData: Ecs) => (ecsRowData?.kibana?.alert ? ecsRowData?._id : null);
```

and then just call it:

```ts
const alertId = getAlertId(ecsRowData);
```

With that change, the `eslint-disable` is no longer required.

Reply:
Thanks, will be in the next push!